File: 3657-check-if-grid-can-be-cut-into-sections/check-if-grid-can-be-cut-into-sections.cpp
class Solution {
public:
    bool checkValidCuts(int n, vector<vector<int>>& rectangles) {
        int m = rectangles.size();
        vector<tuple<int,int>> xdim;
        vector<tuple<int,int>> ydim;
        

        for (const auto &rectangle: rectangles){
            auto xs = rectangle[0];
            auto ys = rectangle[1];
            auto xe = rectangle[2];
            auto ye = rectangle[3];
            xdim.push_back({xs,xe});
            ydim.push_back({ys,ye});
        }
        sort(xdim.begin(),xdim.end());
        sort(ydim.begin(),ydim.end());

        int count = 0;
        int end = get<1>(xdim[0]);
        for (int i = 1 ; i < m; ++i){
            if (get<0>(xdim[i]) >= end) ++count;
            end = max(end, get<1>(xdim[i]));
        }
        if (count > 1) return true;

        count = 0;
        end = get<1>(ydim[0]);
        for (int i = 1 ; i < m; ++i){
            if (get<0>(ydim[i]) >= end) ++count;
            end = max(end, get<1>(ydim[i]));
        }
        if (count > 1) return true;


        return false;
        
    }
};

File: 3657-check-if-grid-can-be-cut-into-sections/check-if-grid-can-be-cut-into-sections.py
class Solution:
    def checkValidCuts(self, n: int, rectangles: List[List[int]]) -> bool:
        

        xdim = []
        ydim = []
        for xs, ys, xe, ye in rectangles:
            xdim.append((xs,xe))
            ydim.append((ys,ye))

        xdim.sort()
        ydim.sort()
        count = 0
        m = len(xdim)
        end = xdim[0][1]
        for i in range(1,m):
            if xdim[i][0] >= end:
                count += 1
            end = max(end, xdim[i][1])
        if count >= 2:
            return True

        end = ydim[0][1]
        count = 0
        for i in range(1,m):
            if ydim[i][0] >= end:
                count += 1
            end = max(end, ydim[i][1])
        if count >= 2:
            return True

        return False

        
